dask/array/core.py

```diff
@@ -1080,6 +1083,50 @@ def numblocks(self):
     def npartitions(self):
         return reduce(mul, self.numblocks, 1)
 
+    def compute_chunk_sizes(self):
+        """
+        Compute the chunk sizes for a Dask array. This is especially useful
+        when the chunk sizes are unknown (e.g., when indexing one Dask array
+        with another).
+
+        Notes
+        -----
+        This function modifies the Dask array in-place.
+
+        Examples
+        --------
+        >>> import dask.array as da
+        >>> import numpy as np
+        >>> x = da.from_array([-2, -1, 0, 1, 2], chunks=2)
+        >>> x.chunks
+        ((2, 2, 1),)
+        >>> y = x[x <= 0]
+        >>> y.chunks
+        ((nan, nan, nan),)
+        >>> y = y.compute_chunk_sizes()
```

Comment:
Maybe just show `y` here, without reassigning, to emphasize that this is inplace.